zerver/models.py

```diff
@@ -1506,14 +1506,26 @@ def __str__(self) -> str:
         return "<%s: %s / %s / %s>" % (self.__class__.__name__, display_recipient,
                                        self.subject, self.sender)
 
+class ArchiveTransaction(models.Model):
+    archive_timestamp = models.DateTimeField(default=timezone_now, db_index=True)  # type: datetime.datetime
+    # Marks if the data archived in this transaction has been restored:
+    restored = models.BooleanField(default=False, db_index=True)  # type: bool
+
+    type = models.PositiveSmallIntegerField(db_index=True)  # type: int
+    # Valid types:
+    RETENTION_POLICY_BASED = 1  # Archiving was executed due to automated retention policies
+    MANUAL = 2  # Archiving was run manually, via move_messages_to_archive function
```

Comment:
Presumably we'll want a type for `MESSAGE_DELETED`, as well?

Reply:
If I understand correctly, I think MANUAL Is your MESSAGE_DELETED? MANUAL signifies it was archived through move_messages_to_archive - this is the function that do_delete_messages functions call. So I think that's the same?

Reply:
I guess this is an integer, so easy to change later, but I think `DELETED_BY_USER` would be a lot clearer.